### DISCUSSION TITLE
Questions

### DISCUSSION THREAD

**Discussion by FuzzieKiwi** [jac0b-w/album-art-wallpaper#18] [Q&A] [answered]
Awesome app. I haven't had any bugs so far. 

I have a few questions:

1. I have it currently setup with Spotify. Is it pulling art from spotify itself or from last.fm? I noticed that some album art is either lower resolution or appears to look compressed. 

I tested Marco Pixel's monstercat-visualizer rainmeter cover art skin and at the same size the album art is clearer. I'm running a 1440P display and also decreased cover art size.

2. If it is compression, would it be possible to have maybe a high res and medium res options?

3. Do you have a PayPal link I can donate to??? :) It's not much but I really appreciate the work you put into this.

**Comment by jac0-b** [comment, marked as answer]
So the art is pulled from whichever service you're using (a 640px jpeg for spotify and I think 600px for last.fm) but it's low res and a jpeg so you still get compression artifacts unfortunately. If I can find a decent API to fetch higher quality album art instead it's definitely something I would like to add (although I am super busy at the moment so this is unlikely to happen for a while) or failing that possibly an AI upscaling option.

I've added a buymeacoffee link in the readme :) glad you like the app the support is much appreciated

**Reply by FuzzieKiwi** [reply]
Awesome and interesting to know. I wish I could help but I'm not good at programming, only know the very basics. I wonder how Marco Pixel's plug-in is obtaining cover art. Regardless, thanks for answering my questions.

**Reply by jac0-b** [reply]
I did try finding the code that gathers the art but wasn't able to find it. If anyone has a link to some code I'll look into it.

**Reply by FuzzieKiwi** [reply]
It uses WebNowPlaying:

https://github.com/tjhrulz/WebNowPlaying

https://github.com/tjhrulz/WebNowPlaying/blob/master/WebNowPlaying.cs
